Sources/Workspace/Workspace.swift

```diff
@@ -1656,17 +1660,16 @@ extension Workspace {
                         guard archiveChecksum == checksum else {
                             tempDiagnostics.emit(.artifactInvalidChecksum(targetName: artifact.targetName, expectedChecksum: checksum, actualChecksum: archiveChecksum))
                             tempDiagnostics.wrap { try self.fileSystem.removeFileTree(archivePath) }
-                            group.leave()
                             return
                         }
 
+                        group.enter()
                         self.archiver.extract(from: archivePath, to: parentDirectory, completion: { extractResult in
```

Comment:
one last thing: add a `defer { group.leave() }` here instead of L1679. otherwise 👍

Reply:
sorry @krzyzanowskim still need to address this little bit

Reply:
done